tests/pipelines/test_api.py

```diff
@@ -192,3 +143,74 @@ def test_delete(get_credentials_mock, pipelines_api):
     assert delete_mock.call_args[0][0] == PIPELINE_ID
     assert delete_mock.call_args[0][1] == CREDENTIALS
     assert delete_mock.call_args[0][2] == HEADERS
+
+
+def test_partition_local_remote(pipelines_api):
+    libraries = [
+        LibraryObject('jar', '/absolute/path/abc.ext'),
+        LibraryObject('jar', 'relative/path.ext'),
+        LibraryObject('jar', 'file:///file/scheme/abs/path.ext'),
+        LibraryObject('jar', 'file://file/scheme/two/slashes/path.ext'),
+        LibraryObject('jar', 'FILE:/all/caps.ext'),
+        LibraryObject('jar', 'FiLe:/weird/case.ext'),
+        LibraryObject('jar', 'file.ext'),
+        # remote
```

Comment:
instead of remote, I'd say shouldn't be uploaded, and the top can say should be uploaded.

Reply:
Changed to remote, I'm commenting the top as "local files" because should be uploading is a bit misleading, as we only upload the files that don't already exist.